### PR TITLE
update scala version to Scala 2.12

### PR DESCRIPTION
**Pull request by rajcspsg**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,8 @@ node_modules/
 .idea
 diagram.png
 slides/
+.idea
```
Comment (stanch) [inline .gitignore:6]: Duplicate?

File: build.sbt [hunk 2]
```diff
@@ -113,8 +113,8 @@ val site = project.in(file("site"))
       file("images/tree+zipper.gif") → "images/tree+zipper.gif",
       ((crossTarget in demoJS).value / "demo-opt.js") → "js/demo.js"
     ),
-    SiteScaladocPlugin.scaladocSettings(config("jvm"), mappings in (Compile, packageDoc) in coreJVM, "api/jvm"),
-    SiteScaladocPlugin.scaladocSettings(config("js"), mappings in (Compile, packageDoc) in coreJS, "api/js"),
+    SiteScaladocPlugin.scaladocSettings( { val Jvm = config("jvm"); Jvm }, mappings in (Compile, packageDoc) in coreJVM, "api/jvm"),
```
Comment (stanch) [inline build.sbt:116]: Could you elaborate on the this?

**Review by stanch — CHANGES_REQUESTED** [review]
Thanks! Looking good. Just a couple of minor things.

File: build.sbt (hunk 2 above)
Comment (rajcspsg) [reply build.sbt:116]: @stanch I faced this issue https://github.com/sbt/librarymanagement/issues/134

File: .gitignore (hunk 1 above)
Comment (rajcspsg) [reply .gitignore]: removed the duplicate entry

File: build.sbt (hunk 2 above)
Comment (stanch) [reply build.sbt:116]: I see! Tricky one :)

**Comment by stanch** [comment]
@rajcspsg FYI, getting some errors: https://travis-ci.org/github/stanch/reftree/builds/772075270

**Comment by rajcspsg** [comment]
@stanch - I created https://github.com/stanch/reftree/pull/29 to fix this

**Pull request merged**